contracts/RewardPool.sol

```diff
@@ -92,13 +92,39 @@ contract RewardPool is Initializable, Versionable, Ownable {
     return true;
   }
 
+  function valid(
+    bytes calldata leaf,
+    bytes calldata proof
+  ) external view returns (bool) {
+    uint256 paymentCycleNumber = uint256(leaf[4]);
+    uint256 root = payeeRoots[paymentCycleNumber];
+    return proof.verify(root, keccak256(leaf));
+  }
+
+  function claimed(leaf) external view returns (bool) {
+    uint256 paymentCycleNumber = uint256(leaf[4]);
```

Comment:
think you meant to use `abi.decode()` here.

Reply:
Looks like we'd need to swap out a packed encoding for a non-packed one as we can't do abi.decodepacked (doesn't exist).
If that's the case, and it has to be a bytes32 array, then because addresses are 20 bytes we'd have lots of space in the "address" bytes32 to store info about the contract (such as whether it's an nft or not).

Clear cost of increased size of data & the hashing though.